### PR TITLE
✨ Deleted all but active sessions on password change

### PR DESCRIPTION
**Pull request by allouis**
refs #10323 

This changes the model to delete ALL sessions for a user from the session store when their password is changed

The session middleware is updated to "resave" the session at end of each request - this keeps the active session alive when changing passwords

### CODE REVIEW

File: core/test/acceptance/admin/users_spec.js [hunk 1]
```diff
@@ -329,5 +329,10 @@ describe('User API', function () {
                 should.exist(res.body.password);
                 should.exist(res.body.password[0].message);
             });
+
+        await request
```
Comment (allouis) [inline core/test/acceptance/admin/users_spec.js]: Even if we stick with this method - we have a test! so we won't accidentally break it

File: core/server/services/permissions/providers.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ var _ = require('lodash'),
 
 module.exports = {
     user: function (id) {
-        return models.User.findOne({id: id, status: 'all'}, {withRelated: ['permissions', 'roles', 'roles.permissions']})
+        return models.User.findOne({id: id, status: 'active'}, {withRelated: ['permissions', 'roles', 'roles.permissions']})
```
Comment (kevinansfield) [inline core/server/services/permissions/providers.js]: Can this go in as a separate commit/PR? It's a different issue to the session invalidation on password change

File: core/server/services/auth/session/middleware.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ const getSession = (req, res, next) => {
         UNO_SESSIONIONA = session({
             store: new SessionStore(models.Session),
             secret: settingsCache.get('session_secret'),
-            resave: false,
+            resave: true,
```
[the remark this answers is not in the document]
Comment (kevinansfield) [reply core/server/services/auth/session/middleware.js]: I agree that this seems too broad. I'd prefer something specific in the change password method as that's the point where the sessions that should be removed is decided

File: core/server/services/permissions/providers.js (hunk 2 above)
Comment (allouis) [reply core/server/services/permissions/providers.js]: Yup this whole thing will be split into just 2 commits!

File: core/server/services/auth/session/middleware.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (allouis) [reply core/server/services/auth/session/middleware.js]: Yeah - the trouble with that is getting current session info into the model layer, which doesn't seem correct or simple. I quite like that the model destroys all of the sessions and that the controller manually intervenes :thinking:

[the remark this answers is not in the document]
Comment (vikaspotluri123) [reply core/server/services/auth/session/middleware.js]: Can you use [`req.session.save()`](https://github.com/expressjs/session#sessionsavecallback) in the controller?
